File: p509e/fib.js
/**
 * @param {number} n
 * @return {number}
 */
var fib = function (n) {
  if (n <= 1) return n;
  let f = [0, 1];
  for (let i = 2; i <= n; ++i) f = [f[1], f[0] + f[1]];
  return f[1];
};
for (let n = 0; n <= 30; ++n) {
  console.log('F(' + n + ') =', fib(n));
}

File: p509e/fib.py
class Solution:
    def fib(self, n: int) -> int:
        if n <= 1:
            return n
        fib = (0, 1)
        for n in range(2, n + 1):
            fib = fib[1], fib[0] + fib[1]
        return fib[1]


# TESTS
for n, expected in [
    (0, 0),
    (1, 1),
    (2, 1),
    (3, 2),
    (4, 3),
    (10, 55),
    (15, 610),
    (30, 832040),
]:
    sol = Solution()
    actual = sol.fib(n)
    print(f"F({n}) = {actual}")
    assert actual == expected
